lte/gateway/c/oai/common/log.c

```diff
@@ -561,10 +561,14 @@ static void log_get_elapsed_time_since_start(struct timeval * const elapsed_time
 //------------------------------------------------------------------------------
 static char* log_get_readable_cur_time(time_t* cur_time) {
   // get the current local time
-  *cur_time = time(NULL);
+  time(cur_time);
+  struct tm* cur_local_time;
+  cur_local_time = localtime(cur_time);
+  char* time_str = calloc(32, sizeof(char));
```

Comment:
This array can be allocated on stack of caller to avoid dynamic memory allocation. Caller can pass explicit array pointer. is there any downside of this ?

Reply:
Yes @pshelar, that is also a feasible approach. We will have to pass the array pointer as well as the array length to the function, increasing the number of arguments to the function. Alternatively, we can fix the size of the time_str using a macro and just use that everywhere. What do you think?

Reply:
Modified in https://github.com/magma/magma/pull/4240/commits/744d161752af1ba0b9da59c88597f3fe9ecc883c